sympy/polys/polytools.py

```diff
@@ -3816,6 +3817,130 @@ def nth_power_roots_poly(f, n):
 
         return r.replace(t, x)
 
+    def root_multiplicity(f, root, prec=10):
+        """
+        Computes the multiplicity of a root for f, at a given
+        numerical precision. Works by evaluating derivatives of
+        f at the given root. If it is not a root of f (up to
+        numerical precision), returns zero.
```

Comment:
I don't think that this function is needed in this context because we will already factorise the polynomial into irreducibles or at least square free parts so the multiplicity is already handled.

Also factorisation is a better method than numerical evaluation for this. Only exact factorisation can reveal multiple roots reliably.